okapi-core/src/main/java/org/folio/okapi/util/ModuleUtil.java

```diff
@@ -167,4 +168,37 @@ public static String moduleList(List<ModuleDescriptor> moduleDescriptors) {
     }
     return str.toString();
   }
+
+  /**
+   * Return list of obsolete modules.
+   *
+   * <p>A module is considered obsolete if it's a snapshot *and* there is a higher
+   * non-snapshot version.
```

Comment:
This assumes that we maintain only the branch of the highest version.
We might maintain a branch for Iris with bugfixes while maintaining a branch for Juniper.

My suggestion:
A module is considered obsolete if it's a snapshot and
a) a higher snapshot exists that differs by the snapshot version only (compareTo is -1 or 1), or
b) at least five higher non-snapshot versions exist.

Reply:
Yes. I was not aware of snapshots in use for bug fix releases or branches therein. That changes the picture , quite a lot. So IMHO we should just not bother with this feature and continue to have all modules in one pile.